// ts-paths/lib/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = require("chalk");
var LevelMap;
(function (LevelMap) {
    LevelMap[LevelMap["ERROR"] = 0] = "ERROR";
    LevelMap[LevelMap["WARN"] = 1] = "WARN";
    LevelMap[LevelMap["INFO"] = 2] = "INFO";
    LevelMap[LevelMap["DEBUG"] = 3] = "DEBUG";
})(LevelMap || (LevelMap = {}));
class Logger {
    set debug(debug) {
        this._debug = debug;
    }
    set level(level) {
        this._level = level;
    }
    info(message) {
        this._print(message, chalk_1.default.green, LevelMap.INFO);
    }
    warn(message) {
        this._print(message, chalk_1.default.yellow, LevelMap.WARN);
    }
    error(message) {
        this._print(message, chalk_1.default.red, LevelMap.ERROR);
    }
    _print(message, color, level) {
        if (!(this._debug && this._level >= level))
            return;
        process.stdout.write(color(`[ts-paths] - `));
        process.stdout.write(color(`${new Date(Date.now()).toLocaleString()}   `));
        process.stdout.write(color(message));
        process.stdout.write(`\n`);
    }
}
exports.Logger = Logger;
